controller/program/tests/tests/set_sol_value_calculator.rs

```diff
@@ -0,0 +1,584 @@
+
+use solana_account::Account;
+use solana_instruction::Instruction;
+use solana_pubkey::Pubkey;
+
+use crate::common::{
+    jupsol_fixtures_svc_suf, max_sol_val_no_overflow, should_fail_with_inf1_ctl_prog_err,
+    should_fail_with_program_err, MAX_LAMPORTS_OVER_SUPPLY, MAX_LST_STATES, SVM,
+};
+
+type SetSolValueCalculatorKeysBuilder =
+    SetSolValueCalculatorIxAccs<[u8; 32], SetSolValueCalculatorIxPreKeysOwned, SvcCalcAccsAg>;
+
+fn set_sol_value_calculator_ix_pre_keys_owned(
+    admin: [u8; 32],
+    token_program: &[u8; 32],
+    mint: [u8; 32],
+) -> SetSolValueCalculatorIxPreKeysOwned {
+    NewSetSolValueCalculatorIxPreAccsBuilder::start()
+        .with_admin(admin)
+        .with_lst_mint(mint)
+        .with_pool_state(POOL_STATE_ID)
+        .with_pool_reserves(find_pool_reserves(token_program, &mint).0.to_bytes())
+        .with_lst_state_list(LST_STATE_LIST_ID)
+        .build()
+}
+
+fn set_sol_value_calculator_ix(
+    builder: &SetSolValueCalculatorKeysBuilder,
+    lst_idx: u32,
+) -> Instruction {
+    let accounts = keys_signer_writable_to_metas(
+        set_sol_value_calculator_ix_keys_owned(builder).seq(),
+        set_sol_value_calculator_ix_is_signer(builder).seq(),
+        set_sol_value_calculator_ix_is_writer(builder).seq(),
+    );
+    Instruction {
+        program_id: Pubkey::new_from_array(ID),
+        accounts,
+        data: SetSolValueCalculatorIxData::new(lst_idx).as_buf().into(),
+    }
+}
+
+fn set_sol_value_calculator_fixtures_accounts_opt(
+    builder: &SetSolValueCalculatorKeysBuilder,
+) -> Vec<PkAccountTup> {
+    fixtures_accounts_opt_cloned(
+        set_sol_value_calculator_ix_keys_owned(builder)
+            .seq()
+            .copied(),
+    )
+    .collect()
+}
+
+pub fn assert_correct_set(
+    bef: &[PkAccountTup],
+    aft: &[PkAccountTup],
+    mint: &[u8; 32],
+    expected_new_calc: &[u8; 32],
+) {
+    let [pools, lst_state_lists] = [POOL_STATE_ID, LST_STATE_LIST_ID]
+        .map(|a| acc_bef_aft(&Pubkey::new_from_array(a), bef, aft));
+    let [pool_bef, pool_aft] = pools.each_ref().map(|a| {
+        PoolStatePacked::of_acc_data(&a.data)
+            .unwrap()
+            .into_pool_state()
+    });
+    let [lst_state_list_bef, lst_state_list_aft] = lst_state_lists
+        .each_ref()
+        .map(|a| LstStatePackedList::of_acc_data(&a.data).unwrap());
+    let lst_state_i = lst_state_list_bef
+        .0
+        .iter()
+        .position(|s| s.into_lst_state().mint == *mint)
+        .unwrap();
+    let [lst_state_bef, lst_state_aft] =
+        [lst_state_list_bef, lst_state_list_aft].map(|l| l.0[lst_state_i].into_lst_state());
+
+    assert_eq!(lst_state_bef.mint, *mint);
+    assert_eq!(lst_state_bef.mint, lst_state_aft.mint);
+
+    assert_ne!(
+        lst_state_bef.sol_value_calculator,
+        lst_state_aft.sol_value_calculator
+    );
+    assert_eq!(&lst_state_aft.sol_value_calculator, expected_new_calc);
```

Comment:
this is scope for a different PR but ive been thinking of doing a diffing system that would use `generic-array-struct` (gas) structs similar to https://github.com/igneous-labs/inf-1.5/blob/master/test-utils/src/gen/stake_pools/spl.rs where we maybe have smth like 

```rust
#[derive(Default...)]
pub enum Diff<T> {
    #[default]
    Unchanged, // assert that field did not change between before and after
    Changed(T, T), // assert that field changed from .0 in before to .1 in after
    Pass, // no-op, dont care
} 
```

This will also help assert that we dont accidentally clobber unrelated fields during account updates